main.cpp

```diff
@@ -39,29 +39,251 @@ send me a DM to check your pull request
+    std::cout << "You are trying to spawn too many knobs at once!\n\n";
+}
+
+void Filter::updateKnobPosition(FilterKnob knobToUpdate, double nextKnobPosition)
+{
+    int knobID = knobToUpdate.knobID;
+    std::cout << "The knob to update is: " << knobID << ". The next knob position is: " << nextKnobPosition << ".\n" << std::endl;
+    std::cout << "Your filter frequency is:" << frequency << std::endl;
+}
 /*
  copied UDT 2:
  */
+struct WavetableOscillator
+{
+    double frequency;
+    int waveShape;
+    float volumeLevel;
+
+    WavetableOscillator() : frequency(4000.0), waveShape(2), volumeLevel(1.0f) 
+    {
+        std::cout << "Spawning WavetableOscillator object!\n" << std::endl;
+    }
+    ~WavetableOscillator()
+    {
+        std::cout << "Destroying WavetableOscillator object!\n" << std::endl;
+    }
+
+    struct Wavetable
+    {   
+        int maxNumberOfWaveforms;
+        bool interpolationOn;
+        int currentWavetable;
+
+        Wavetable()
+        {
+            maxNumberOfWaveforms = 100;
+            interpolationOn = true;
+            currentWavetable = 0;  
+        }
+                    
+        int cycleToTargetWavetable(WavetableOscillator::Wavetable table, int targetTable);   
+        void loadWavetable(int wavetableToLoad, float loadTimeOffset);
+    };
 
+    int getNextFrequencyInSequence(float nextFrequency)
+    {
+        std::cout << "The next frequency is: " << nextFrequency << ".\n" << std::endl;
+
+        std::cout << "Waveshape is: " << waveShape << ". Volume Level is: " << volumeLevel << ".\n" << std::endl;
+        return{};
+    }
+
+    float fadeOutVolume(WavetableOscillator osc, float incrementAmount)
+    {
+        for(float i = osc.volumeLevel; i >= 0; i -= incrementAmount)
+        {
+            osc.volumeLevel -= incrementAmount;
+            std::cout << "Current oscillator volume is: " << osc.volumeLevel << "." << std::endl;
+            if(osc.volumeLevel <= 0)
+            {
+                std::cout << "Finished fading out!\n" << std::endl;
+                return osc.volumeLevel;
+            }
+        }
+        
+        return osc.volumeLevel;
+    }
+};
 /*
  copied UDT 3:
  */
+struct Reverb
+{
+    int decayTime;
+    float roomSize;
+    double inputVolume;
+    double outputVolume;
+    double dryWet;
+
+    Reverb() : decayTime(1), roomSize(3.5f), inputVolume(1.0), outputVolume(1.0), dryWet(1.0) 
+    {
+        std::cout << "Spawning Reverb object!\n" << std::endl;
+    }
+
+    ~Reverb()
+    {
+        std::cout << "Destroying Reverb object!\n" << std::endl;
+    }
+
+    int setNextReverbSettings(Reverb newVerbSettings);
+
+    double setNextDryWet(Reverb targetVerb, double nextDryWetValue)
+    {
+        double epsilon = 0.0001;
+
+        std::cout << "\nAdjusting reverb dryWet!" << std::endl;
+        double incrementPolarity;
+
+        if(targetVerb.dryWet < nextDryWetValue) incrementPolarity = 0.1;
+        else incrementPolarity = -0.1;
```

Comment:
Your `targetVerb` is being used in place of the instance that is having `setNextDryWet()` called on it. 
- [ ] remove `Reverb targetVerb` from parameter list
- [ ] revise function to use member variables of the calling instance.